internal/injector/builtin/yaml/logs/logrus.yml

```diff
@@ -0,0 +1,89 @@
+meta:
+  name: github.com/sirupsen/logrus
+  description: Structured, pluggable logging for Go.
+  icon: at-symbol
+aspects:
+  - id: Inject DDContextLogHook
+    join-point:
+      struct-definition: github.com/sirupsen/logrus.Logger
+    advice:
+      - inject-declarations:
+          imports:
+            telemetry: gopkg.in/DataDog/dd-trace-go.v1/internal/telemetry
+            tracer: gopkg.in/DataDog/dd-trace-go.v1/ddtrace/tracer
+            ext: gopkg.in/DataDog/dd-trace-go.v1/ddtrace/ext
+          template: |-  
+            func init() {
+              telemetry.LoadIntegration("sirupsen/logrus")
+              tracer.MarkIntegrationImported("github.com/sirupsen/logrus")
+            }
+            
+            // DDContextLogHook ensures that any span in the log context is correlated to log output.
+            type DDContextLogHook struct{}
+            
+            // Levels implements logrus.Hook interface, this hook applies to all defined levels
+            func (d *DDContextLogHook) Levels() []Level {
+              return []Level{PanicLevel, FatalLevel, ErrorLevel, WarnLevel, InfoLevel, DebugLevel, TraceLevel}
+            }
+            
+            // Fire implements logrus.Hook interface, attaches trace and span details found in entry context
+            func (d *DDContextLogHook) Fire(e *Entry) error {
+              span, found := tracer.SpanFromContext(e.Context)
+              if !found {
+                return nil
+              }
+              e.Data[ext.LogKeyTraceID] = span.Context().TraceID()
+              e.Data[ext.LogKeySpanID] = span.Context().SpanID()
+              return nil
+            }
+
+  - id: Trace logrus.New
+    join-point:
+      all-of:
+        - import-path: github.com/sirupsen/logrus
+        - function-body:
+            function:
+              - name: New
+    advice:
+      - prepend-statements:
+          template: |-
+            {{- $logger := .Function.Result 0 -}}
+            defer func() {
+              {{ $logger }}.AddHook(&DDContextLogHook{})
+            }()
+
+  - id: Wrap logrus.Logger (pointer)
+    join-point:
+      struct-literal:
+        type: github.com/sirupsen/logrus.Logger
+        match: pointer-only
+    advice:
+      - wrap-expression:
+          imports:
+            logrus: github.com/sirupsen/logrus
+          template: |-
+            func(logger *logrus.Logger) *logrus.Logger {
+              logger.AddHook(&logrus.DDContextLogHook{})
+              return logger
+            }({{ . }})
+
+  - id: Wrap logrus.Logger (value)
+    join-point:
+      struct-literal:
+        type: github.com/sirupsen/logrus.Logger
+        match: value-only
+    advice:
+      - wrap-expression:
+          imports:
+            logrus: github.com/sirupsen/logrus
+          template: |-
+            func(logger logrus.Logger) logrus.Logger {
+              logger.AddHook(&logrus.DDContextLogHook{})
+              return logger
+            }({{ . }})
```

Comment:
I don't love how this appears to be making more copies than strictly necessary; but I reckon I might be overly pessimistic with how the compiler will treat that, and I don't think this is a line that ought to be on a hot path anyway...